package ProtoMission;

message MissionCPP
{
	required uint32 id			= 1;  //任务id
	required bool status		= 2;  //任务状态
	repeated uint32 current		= 3;   //任务当前进度
}

message MissionAllCPP
{
	repeated MissionCPP missions	= 1;
}

//推送任务的请求
message PushMissionReq
{
	required MissionAllCPP missions		= 1;
}

//验证国战任务的请求
message VerfifyMissonReq
{

}

//验证世界移动任务的请求
message VerifyWorldMoveMissionReq
{
	
}

//领取任务奖励
message GetMissionRewardReq
{
	required uint32 id		= 1;
}

message GetMissionRewardResp
{
	required uint32 food	= 1;   //粮食
	required uint32 coin	= 2;   //银两
	required uint32 wood	= 3;   //木材
	required uint64 exp		= 4;   //用户经验
	required uint32 level	= 5;    //用户等级
	required MissionAllCPP nextmission	= 6;   //下一批解锁的任务
	optional uint32 hammer	= 7;	//黄金锤子
}
